src/backend/plaf.rs

```diff
@@ -0,0 +1,285 @@
+                let first = self.convert_plaf_poly(iter.next().unwrap(), counter);
+                iter.fold(first, |acc, e| acc + self.convert_plaf_poly(e, counter))
+            }
+            cPolyExpr::Mul(es) => {
+                let mut iter = es.iter();
+                let first = self.convert_plaf_poly(iter.next().unwrap(), counter);
+                iter.fold(first, |acc, e| acc * self.convert_plaf_poly(e, counter))
+            }
+            cPolyExpr::Neg(e) => -self.convert_plaf_poly(e, counter),
+            cPolyExpr::Pow(e, n) => {
+                if *n == 0 {
+                    pExpr::Const(BigUint::from(1u32))
+                } else {
+                    let e = self.convert_plaf_poly(e, counter);
+                    (1..*n).fold(e.clone(), |acc, _| acc * e.clone())
+                }
+            }
+            cPolyExpr::Halo2Expr(e) => pExpr::from(e),
+            cPolyExpr::Query(column, rotation, annotation) => {
+                let index = self.c_column_id_to_p_column_index.get(&column.uuid()).unwrap();
+                if self.debug {
+                    println!("GET c column id {} match p column index {}", column.uuid(), index);
+                    println!("MAP {:#?}", self.c_column_id_to_p_column_index);
+                }
+                pExpr::Var(PlonkVar::Query(self.convert_plaf_query(column, rotation, annotation, *index)))
+            }
+        }
+    }
+
+    fn add_id_index_mapping(&self, column: &cColumn, c_column_id_to_p_column_index: Option<&mut HashMap<u32, usize>>, counter: &mut Counter) {
+        match c_column_id_to_p_column_index {
+            Some(c_column_id_to_p_column_index) => {
+                c_column_id_to_p_column_index.insert(column.uuid(), counter.number());
+                if self.debug {
+                    println!("c column id {} match p column index {}", column.uuid(), counter.number());
+                }
+                counter.increment();
+            }
+            None => {}
+        }
+    }
+
+    fn convert_plaf_query(
+        &self,
+        column: &cColumn,
+        rotation: &i32,
+        _annotation: &String,
+        index: usize, // plaf index starts from 0 for each column type
+    ) -> ColumnQuery {
+        match column.ctype {
+            cAdvice | Halo2Advice => {
+                ColumnQuery {
+                    column: pColumn {
+                        kind: ColumnKind::Witness,
+                        index,
+                    }, 
+                    rotation: rotation.clone(),
+                }
+            }
+            cFixed | Halo2Fixed => {
+                ColumnQuery {
+                    column: pColumn {
+                        kind: ColumnKind::Fixed,
+                        index,
+                    }, 
+                    rotation: rotation.clone(),
+                }
+            }
+        }
+    } 
+}
+
+
+#[derive(Clone, Debug)]
+pub struct Counter {
+    number: usize,
+}
+
+impl Counter {
```

Comment:
I don't think we need to wrap an usize in a struct. We can just use the usize.

Reply:
Fixed